data/tmdb-movies/src/main/java/io/fimltime/data/tmdb/movies/TmdbMovieRepositoryImpl.kt

```diff
@@ -94,4 +100,17 @@ internal class TmdbMovieRepositoryImpl @Inject constructor(
 
   override suspend fun getCollection(collectionId: Int): Result<MovieCollection, GeneralError> =
     tmdbMoviesRemoteSource.getCollection(collectionId)
+
+  override suspend fun getBookmarkedMovies(): Flow<List<VideoThumbnail>> {
+    return bookmarksDao.getAllBookmarksByType(Movie).map { bookmarks ->
+      val movieList: MutableList<VideoThumbnail> = mutableListOf()
+      for (id in bookmarks) {
+        when (val result = fetchMovieDetailsFromNetwork(id.tmdbId)) {
+          is Failure -> continue
+          is Success -> movieList.add(result.data.toVideoThumbnail())
+        }
+      }
+      movieList
```

Comment:
Not a good idea to make multiple requests every time user asks for bookmarks. Maybe saving thumbnail image helps here. I'll create an issue for it #119